src/conn/conn_cache_pool.c

```diff
@@ -453,22 +468,41 @@ __cache_pool_assess(WT_SESSION_IMPL *session, uint64_t *phighest)
 			continue;
 		cache = entry->cache;
 		++entries;
-		new = cache->bytes_read;
+		/* Copy the value out so it doesn't change underneath us */
+		tmp = cache->bytes_read;
 		/* Handle wrapping of eviction requests. */
-		if (new >= cache->cp_saved_read)
-			cache->cp_current_read = new - cache->cp_saved_read;
+		if (tmp >= cache->cp_saved_read)
+			cache->cp_pass_read = tmp - cache->cp_saved_read;
 		else
-			cache->cp_current_read = new;
-		cache->cp_saved_read = new;
-		if (cache->cp_current_read > highest)
-			highest = cache->cp_current_read;
+			cache->cp_pass_read = tmp;
+		cache->cp_saved_read = tmp;
+
+		/* Update the application eviction count information */
+		tmp = cache->cp_saved_app_evicts;
+		cache->cp_pass_app_evicts =
+		    tmp - cache->cp_pass_app_evicts;
+		cache->cp_saved_app_evicts = tmp;
+
+		/* Update the eviction wait information */
+		tmp = cache->cp_saved_app_waits;
+		cache->cp_pass_app_waits =
+		    tmp - cache->cp_pass_app_waits;
+		cache->cp_saved_app_waits = tmp;
```

Comment:
This seems to have the same problem: I don't understand what is being stored in "app_waits" vs "pass_waits".